src/inmanta/data/__init__.py

```
@@ -213,6 +234,326 @@ def ensure_boundary_type(
+        self.db_order = db_order
+        self.limit = limit
+        self.backward_paging = backward_paging
+
+    def select(self, select_clause: str) -> "SimpleQueryBuilder":
+        """ Set the select clause of the query """
+        return SimpleQueryBuilder(
+            select_clause,
+            self._from_clause,
+            self.filter_statements,
+            self.values,
+            self.db_order,
+            self.limit,
+            self.backward_paging,
+        )
+
+    def from_clause(self, from_clause: str) -> "SimpleQueryBuilder":
+        """ Set the from clause of the query"""
+        return SimpleQueryBuilder(
+            self.select_clause,
+            from_clause,
+            self.filter_statements,
+            self.values,
+            self.db_order,
+            self.limit,
+            self.backward_paging,
+        )
+
+    def order_and_limit(
+        self, db_order: DatabaseOrder, limit: Optional[int] = None, backward_paging: Optional[bool] = False
+    ) -> "SimpleQueryBuilder":
+        """ Set the order and limit of the query """
+        return SimpleQueryBuilder(
+            self.select_clause, self._from_clause, self.filter_statements, self.values, db_order, limit, backward_paging
+        )
+
+    def filter(self, filter_statements: List[str], values: List[object]) -> "SimpleQueryBuilder":
+        return SimpleQueryBuilder(
+            self.select_clause,
+            self._from_clause,
+            self.filter_statements + filter_statements,
+            self.values + values,
+            self.db_order,
+            self.limit,
+            self.backward_paging,
+        )
+
+    def build(self) -> Tuple[str, List[object]]:
+        if not self.select_clause:
+            raise InvalidQueryParameter("A valid query must have a SELECT clause")
+        full_query = f"""{self.select_clause}
+                         {self._from_clause if self._from_clause else ""}
```

A SQL query without a FROM clause is not going to work, is it? Or am I missing something?

---

It's technically allowed by PostgreSQL, but you're probably right that it's such a particular use case that in most cases it's better to raise an exception (and in case someone needs a query that allows not setting it, that can be another subclass). I'll make the changes